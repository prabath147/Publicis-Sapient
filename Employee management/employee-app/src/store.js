import { createStore, applyMiddleware } from "redux";
import { composeWithDevTools } from "redux-devtools-extension";
import rootReducer from "./Component/reducers";
import Thunk from "redux-thunk";

const store = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(Thunk))
);

export default store;
