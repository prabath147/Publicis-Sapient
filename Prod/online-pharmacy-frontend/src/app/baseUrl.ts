export const baseURL = "https://frk5iphcfk.execute-api.us-east-1.amazonaws.com/";
    // "http://localhost:8081";
    
