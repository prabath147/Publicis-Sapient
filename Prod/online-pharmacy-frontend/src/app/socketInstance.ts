
export const socketURL = "https://c8zpvs7fv4.execute-api.us-east-1.amazonaws.com/notify-ws"

// const socket = new SockJS(socketURL);
// export const stompClient = Stomp.over(socket);