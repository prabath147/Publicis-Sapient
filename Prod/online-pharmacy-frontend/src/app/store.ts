import { Action, configureStore, ThunkAction } from "@reduxjs/toolkit";
import StoreReducer from "../components/pages/Admin/storeList/StoreSlice";
import SubscriptionReducer from "../components/pages/AdminSubscriptionPack/SubscriptionvSlice";
import UserReducer from "../components/pages/Auth/Login/UserSlice";
import CartReducer from "../components/pages/Cart/CartSlice";
import ManagerReducer from "../components/pages/Manager/ManagerSlice";
import NotificationReducer from "../components/pages/Open/notificationList/NotificationSlice";
// import OrderSlice from "../components/pages/orderDetail/OrderSlice";
// import OrderListSlice from "../components/pages/orderList/OrderListSlice";

export const store = configureStore({
  reducer: {
    storeL: StoreReducer,
    user: UserReducer,
    notification: NotificationReducer,
    viewsubscription: SubscriptionReducer,
    // addsubscription: AddSubcriptionRe,
    cart: CartReducer,
    // order: OrderSlice,
    // orderL: OrderListSlice,
    manager: ManagerReducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false,
    }),
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
