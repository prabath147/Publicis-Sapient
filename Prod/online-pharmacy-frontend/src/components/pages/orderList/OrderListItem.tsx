import { Card, Grid, Text } from "@mantine/core";
import { Link } from "react-router-dom";

export default function OrderListItem({
  id,
  customer,
  orderDate,
  price,
  shippingAddress,
}) {
  // const monthNames = [
  //   "January",
  //   "February",
  //   "March",
  //   "April",
  //   "May",
  //   "June",
  //   "July",
  //   "August",
  //   "September",
  //   "October",
  //   "November",
  //   "December",
  // ];

  return (
    <Card withBorder radius={"md"} px={30} mb={"sm"}>
      <Card.Section withBorder>
        <Grid>
          <Grid.Col span={2}>
            <Text color={"dimmed"} transform="uppercase">
              Order Placed
            </Text>
            <Text>{orderDate.toString()}</Text>
          </Grid.Col>

          <Grid.Col span={2}>
            <Text align="center" color={"dimmed"} transform="uppercase">
              Total
            </Text>
            <Text align="center">{price}</Text>
          </Grid.Col>

          <Grid.Col span={2}>
            <Text color={"dimmed"} transform="uppercase">
              Ship to
            </Text>
            <Text>{customer}</Text>
          </Grid.Col>

          <Grid.Col span={3}>
            <Text color={"dimmed"} transform="uppercase">
              Shipping Address
            </Text>
            <Text lineClamp={2}>{
              `${customer}, ${shippingAddress.street} ${shippingAddress.city} ${shippingAddress.state} ${shippingAddress.country} ${shippingAddress.pinCode}`
            }</Text>
          </Grid.Col>

          <Grid.Col span={2}>
            <Link to={"/orders/" + id}>
              ORDER # {id}
            </Link>
          </Grid.Col>
        </Grid>
      </Card.Section>
    </Card>
  );
}
