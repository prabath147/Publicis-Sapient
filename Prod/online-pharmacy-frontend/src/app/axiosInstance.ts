import axios, { AxiosRequestConfig } from "axios";
import LocalUserService from "../services/LocalUserService";
import TokenService from "../services/TokenService";
import { baseURL } from "./baseUrl";

const axiosInstance = axios.create({
  baseURL: baseURL,
  timeout: 5000,
  headers: {
    "Content-Type": "application/json",
    accept: "application/json",
  },
});

axiosInstance.interceptors.request.use(
  (config: AxiosRequestConfig) => {
    const token = TokenService.getLocalAccessToken();
    if (token && config.headers !== undefined) {
      config.headers["Authorization"] = "Bearer " + token; // for Spring Boot back-end
      // config.headers["x-access-token"] = token; // for Node.js Express back-end
      config.headers["ngrok-skip-browser-warning"] = "";
    }
    return config;
  },
  (error) => {
    return Promise.reject(error);
  }
);

axiosInstance.interceptors.response.use(
  (res) => {
    return res;
  },
  async (err) => {
    const originalConfig = err.config;
    if (originalConfig.url !== "api/auth/signin" && err.response) {
      // Access Token was expired
      if (err.response.status === 401 && !originalConfig._retry) {
        originalConfig._retry = true;
        try {
          const rs = await axiosInstance.post("api/auth/refresh-token", {
            refreshToken: TokenService.getLocalRefreshToken(),
          });
          const { accessToken } = rs.data;
          TokenService.updateLocalAccessToken(accessToken);
          return axiosInstance(originalConfig);
        } catch (_error) {
          TokenService.removeLocalAccessToken()
          TokenService.removeLocalRefreshToken()
          LocalUserService.removeLocalUser()
          // window.location.reload();
          return Promise.reject(_error);
        }
      }
    }
    return Promise.reject(err);
  }
);
export default axiosInstance;
