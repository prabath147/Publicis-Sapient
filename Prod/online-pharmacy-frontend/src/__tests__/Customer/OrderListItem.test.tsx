
import dayjs from 'dayjs';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { store } from '../../app/store';
import OrderListItem from '../../components/pages/orderList/OrderListItem';

it('component created', () => {
    const component = renderer.create(

        <Provider store={store}>
            <BrowserRouter>
                <OrderListItem id={''} orderDate={dayjs(new Date("2022-01-10")).add(3, "days").toDate()} price={0} customer={''} shippingAddress={''} />
            </BrowserRouter>
        </Provider>
    )
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})
