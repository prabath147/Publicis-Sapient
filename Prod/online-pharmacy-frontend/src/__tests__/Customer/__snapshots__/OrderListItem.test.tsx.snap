// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component created 1`] = `
<div
  className="mantine-Paper-root mantine-Card-root mantine-nanwuu"
>
  <div
    className="mantine-Card-cardSection mantine-sse0oy"
    data-first={true}
    data-last={true}
    padding="md"
  >
    <div
      className="mantine-Grid-root mantine-nsdj"
    >
      <div
        className="mantine-Col-root mantine-1vwjzh"
      >
        <div
          className="mantine-Text-root mantine-1mw9aa3"
        >
          Order Placed
        </div>
        <div
          className="mantine-Text-root mantine-1pbxw0k"
        >
          Thu Jan 13 2022 05:30:00 GMT+0530 (India Standard Time)
        </div>
      </div>
      <div
        className="mantine-Col-root mantine-1vwjzh"
      >
        <div
          className="mantine-Text-root mantine-1qjc1bw"
        >
          Total
        </div>
        <div
          className="mantine-Text-root mantine-1pe3xtm"
        >
          0
        </div>
      </div>
      <div
        className="mantine-Col-root mantine-1vwjzh"
      >
        <div
          className="mantine-Text-root mantine-1mw9aa3"
        >
          Ship to
        </div>
        <div
          className="mantine-Text-root mantine-1pbxw0k"
        />
      </div>
      <div
        className="mantine-Col-root mantine-1ddzqhw"
      >
        <div
          className="mantine-Text-root mantine-1mw9aa3"
        >
          Shipping Address
        </div>
        <div
          className="mantine-Text-root mantine-b62wm9"
        >
          , undefined undefined undefined undefined undefined
        </div>
      </div>
      <div
        className="mantine-Col-root mantine-1vwjzh"
      >
        <a
          href="/orders/"
          onClick={[Function]}
        >
          ORDER # 
        </a>
      </div>
    </div>
  </div>
</div>
`;
